src/natcap/invest/spec_utils.py

```diff
@@ -613,26 +616,56 @@ def describe_arg_from_name(module_name, *arg_keys):
     return f'.. _{anchor_name}:\n\n{rst_description}'
 
 
-def write_metadata_file(datasource_path, spec, lineage_statement, keywords_list):
-    """Write a metadata sidecar file for an invest output dataset.
+def write_metadata_file(datasource_path, spec, keywords_list,
+                        lineage_statement='', out_workspace=None):
+    """Write a metadata sidecar file for an invest dataset.
+
+    Create metadata for invest model inputs or outputs, taking care to
+    preserve existing human-modified attributes.
+
+    Note: We do not want to overwrite any existing metadata so if there is
+    invalid metadata for the datasource (i.e., doesn't pass geometamaker
+    validation in ``describe``), this function will NOT create new metadata.
+    Instead, if ``out_workspace`` specified, the existing metadata will be
+    copied here.
 
     Args:
-        datasource_path (str) - filepath to the invest output
-        spec (dict) -  the invest specification for ``datasource_path``
-        lineage_statement (str) - string to describe origin of the dataset.
+        datasource_path (str) - filepath to the data to describe
+        spec (dict) - the invest specification for ``datasource_path``
         keywords_list (list) - sequence of strings
-
+        lineage_statement (str, optional) - string to describe origin of
+            the dataset
+        out_workspace (str, optional) - where to write metadata if different
+            from data location
     Returns:
-        None
+        None: if metadata could not be created due to validation or file errors.
```

Comment:
I think technically it returns `None` in all cases. A function with no `return` statement will return `None`.